cfy_manager/components/validations.py

```diff
@@ -417,6 +417,26 @@ def _validate_ssl_and_external_certificates_match():
                               "`ssl_enabled` to true.")
 
 
+def _is_cert_self_signed(cert_file):
+    cmd = ['openssl', 'verify', '-CAfile', cert_file, cert_file]
+    output = subprocess.Popen(cmd, stdout=subprocess.PIPE).communicate()[0]
+    if output[-3:-1] == 'OK':
```

Comment:
Can't you just use returncode == 0? Then you don't need to do any output wrangling.

(e.g. return output.returncode == 0)

Reply:
I thought about it also, but the thing is that when you put a CA-signed certificate, the return code is 512 and I couldn't find anything on the internet to verify that this would happen all the time. Is there a convention that any return code other than 0 means error? If yes, I'll change it.

Reply:
Yes, in well behaved shell applications. I believe openssl is one of those.

Reply:
Side note: It's possible that 512 is a specific code they use to indicate that it's signed by another cert, but I don't think that matters for this function?

Reply:
Great thanks, I'll change it and update the PR.

Reply:
Done